x-pack/lib/monitoring/monitoring.rb

```diff
@@ -69,8 +71,12 @@ def auth?
         user && password
       end
 
+      def api_key?
+        api_key
+      end
+
       def ssl?
-        ca_path || (truststore_path && truststore_password) || (keystore_path && keystore_password)
+        ssl || ca_path || (truststore_path && truststore_password) || (keystore_path && keystore_password)
```

Comment:
since we instantiate this plugin with `'ssl' => true` in the same conditions, is the duplication of logic desirable?

Reply:
semantically this implementation is correct since an `ssl => true` option *could* be passed without any certificate related options so I think it is better to also accept that here.